addon/serializer.js

```diff
@@ -0,0 +1,37 @@
+import Model from './orm/model';
+import Collection from './orm/collection';
+import { pluralize } from './utils/inflector';
+import extend from './utils/extend';
+
+class Serializer {
+
+  serialize(response, request) {
+    if (response instanceof Model) {
+      return this.serializeModel(response);
+
+    } else if (response instanceof Collection) {
+      return this.serializeCollection(response);
+
+    } else {
+      return response;
+    }
+  }
+
+  serializeModel(model) {
+    return this.constructor.root ? { [model.type]: model.attrs } : model.attrs;
```

Comment:
Just highlight that this patterns only applies to AMS-like serializers. 
JSONAPI serializer never include the root (or always include the root `data`, depends on how you interpret it). If this class intends to be a superclass of other serializers maybe it's not generic enough.